apps/frontend/src/app/components/user-search/user-search.component.ts

```diff
@@ -0,0 +1,95 @@
+import {
+  Component,
+  ElementRef,
+  EventEmitter,
+  OnInit,
+  Output,
+  ViewChild
+} from '@angular/core';
+import { FormBuilder, FormGroup } from '@angular/forms';
+import { UsersService } from '@momentum/frontend/data';
+import { User } from '@momentum/types';
+import { debounceTime, distinctUntilChanged } from 'rxjs/operators';
+
+@Component({
+  selector: 'mom-user-search',
+  templateUrl: './user-search.component.html',
+  styleUrls: ['./user-search.component.scss']
+})
+export class UserSearchComponent implements OnInit {
+  userSearchForm: FormGroup = this.fb.group({ search: [''] });
+  get nameSearch() {
+    return this.userSearchForm.get('search');
+  }
+
+  get isSteamId() {
+    return (
+      typeof this.userSearchForm.get('search').value == 'string' &&
+      this.userSearchForm.get('search').value.startsWith('steam:')
+    );
+  }
+
+  foundUsers: User[];
+
+  pageLimit: number;
+  userSearchCount: number;
+  userSearchPage: number;
+
+  @Output() selectedUserEmit: EventEmitter<User>;
+  @ViewChild('searchInput', { static: true }) searchInput: ElementRef;
+  constructor(private fb: FormBuilder, private usersService: UsersService) {
+    this.selectedUserEmit = new EventEmitter<User>();
+    this.foundUsers = [];
+    this.pageLimit = 5;
+    this.userSearchCount = 0;
+    this.userSearchPage = 1;
+  }
+
+  findUsers(val: string) {
+    val = val.trim();
+    if (val.length > 0)
+      if (val.startsWith('steam:'))
+        this.usersService
+          .getUsers({
+            steamID: val.slice(6),
```

Comment:
Let's only send this query is `val.slice(6)` actually could be a valid steamID64, i.e. is a string of an integer.